### ISSUE TITLE
Consider putting the runtime in a sub folder

### ISSUE THREAD

**Issue by Petermarcu** [dotnet/core-setup#1219]
## Steps to reproduce
Publish a standalone app

## Expected  behavior
My app is in the root and all the runtime goo is hidden in a sub folder

## Actual behavior
I publish my app and I get over 100 binaries that aren't mine. They are just noise for my users who try to launch my app from my folder.

## Environment data

**Comment by gkhanna79** [comment]
The output is a function of publish operation that CLI implements. Secondly, the layout has strong dependencies with the host and where SharedFX can be found.

**Comment by Petermarcu** [comment]
yeah, I put the issue here because I assumed publish was putting it where the host could find. In order to do something like this, I assume we'd have to change the host and then update publish. Is that correct?

**Comment by PurelySimple** [comment]
Agreed that this is annoying for a self contained deployment. I thought about distributing my application with a shortcut to the actual application at an upper level but obviously this is not cross platform compatible. If the application has such a strong dependency on having the .NET Core runtime in the same directory, could we get a publish option that combines the entire runtime into a single DLL? It's not like our end user cares what version of what component of the core runtime we're using anyway.

**Comment by menasheh** [comment]
anything to be done about this?

**Comment by dagood** [comment]
@vitek-karas @swaroop-sridhar is this something on your plate? Is the scenario maybe covered by single-file bundling?

**Comment by vitek-karas** [comment]
If single-file can be used, then it basically solves this problem.
For non-single-file case, this would require SDK and hosting changes which we haven't considered yet.

@menashes I understand the general desire of this issue, but I would be interested in your particular scenario and why is it important to you? Is it that you "don't like it" (which is a valid reason, just asking), or is it that this creates an issue for the users, or something else entirely?

Would this be something you would expect an option for during `dotnet publish`? Or you would like to have other ways to get it?

**Comment by menasheh** [comment]
@vitek-karas I would expect this to be the default behavior for dotnet publish.

We have a .net core console app which we build for Windows and Mac.

The app builds reports for our client services team. The business people expect an executable file and a readme. If there are 8 files in the folder they can probably find those two files easily enough in Finder or Explorer. If there are 235 files in the folder, it is too complicated to use.

Overall, way too much scrolling necessary to get to the files you actually need.
